### PR TITLE
feat: Offline tx signing with CosmJS primitives

### PR DESCRIPTION
**Pull request by jay-dee7**
This PR adds an additional method for offline transaction signing `signAminoTx`, this method is being added as a new method (instead of modifying the existing ones) since their method signatures make it a breaking change. We'll replicate this for all the methods (signTransaction, signMessage, Broadcast or signAndBroadcast) and then phase out the existing lum-network based implementation.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
         "axios": "^0.27.2",
         "bootstrap": "^5.1.2",
         "clipboard": "^2.0.6",
-        "cosmjs-types": "^0.5.0",
+        "cosmjs-types": "0.4.0",
```
Comment (Eengineer1) [inline package.json:22]: @jay-dee7 Is there a reason for the downgrade?

File: src/network/types/msg.ts [hunk 2]
```diff
@@ -34,3 +35,28 @@ export interface DocSigner {
...
+	memo?: string;
+	/**
+	 * Transactions messages
+	 */
+	messages: EncodeObject[];
+	/**
+	 * Transction auth signers
+	 */
+	signers: DocSigner[];
+}
```
Comment (Eengineer1) [inline src/network/types/msg.ts:62]: @jay-dee7 What's the exact reason for this interface definition?

**Review by Eengineer1 — CHANGES_REQUESTED** [review]
@jay-dee7 Looks good up to this point.
Needs a few refactors for clearing up unused code along with accurate types & interface definitions.

Thanks!

File: src/network/types/msg.ts (hunk 2 above)
Comment (jay-dee7) [reply src/network/types/msg.ts:62]: @Eengineer1 I'm just trying to phase the entire LUM-related fields. The regular `Doc` uses `Message` from lum-sdk and I just want to make it cleaner. We'll just reuse this Doc everywhere, rewrite everything with cosmjs and remove the Doc from here

File: package.json (hunk 1 above)
Comment (jay-dee7) [reply package.json:22]: @Eengineer1 that other cosmjs libs that we use (like `proto-signing` or `stargate`) use cosmjs-types@0.4.0 which so using v0.5.0 causes type conflicts. They're switching to v0.5 of cosmjs-types in v0.29 of @cosmjs/* packages, so we can upgrade this then

**Review by Eengineer1 — APPROVED** [review]
@jay-dee7 Looks stable. Approved.

**Pull request merged**